index.js

```diff
@@ -25,7 +25,7 @@ module.exports = class TracingConfigPlugin {
 
   toggleTracing(functionName, isEnabled) {
     this.serverless.cli.log(`Tracing ${isEnabled ? 'ENABLED' : 'DISABLED'} for function "${functionName}"`);
```

Comment:
1. This log message will be misleading as tracing is neither enabled nor disabled when run with `noDeploy`.
Any reason the logic is placed at the function level, and not in the constructor (like in the initial version) or `processTracing(...)`? Maybe it would make more sense to do the return in `processTracing(...)`, for instance:
```
processTracing() {
  if (this.options.noDeploy === true) {
    this.serverless.cli.log(`Serverless is run in "noDeploy" mode, skipping tracing settings`);
    return Promise.resolve('noop'); // make sure to always return a promise
  }
// the rest body of the function
}
```
2. Looks like this feature is merged: https://github.com/serverless/serverless/issues/2660 and it removes the `--nodeploy` command option. Does this affect `this.options.noDeploy`?
3. Which serverless version the change has been tested with?

Reply:
> Which version of serverless the change was tested with?

`sls v.1.20.2`

> Looks like this feature is merged

That sls PR has too many changes.
I was looking into this comment https://github.com/serverless/serverless/issues/1884#issuecomment-240420372
But they have resources to change 100+ files now. Still `--noDeploy` interface works for us well. I guess they support backward compatibility of such interfaces.

> This log message will be misleading

We need a clean run with minimal change to scripts / arguments. Nobody will care if `sls`  is generating templates or executing `aws`. But from the testing standpoint logs show, that this plugin works.

Reply:
Looks like `--noDeploy` works well for you with this version of serverless, so it doesn't look like `this.options.noDeploy` is going away. Let's keep it and not worry about `package` for now.
Okay, looks like there is some benefit in keeping the logs from the testing perspective, let's keep it too.

Let's just wrap the no-op response with `Promise.resolve(...)`, to have consistent return types, make sure `npm t` and linting passes, and should be good to go.